src/v/redpanda/admin_server.cc

```diff
@@ -142,6 +143,22 @@ static ss::logger logger{"admin_api_server"};
 // Helpers for partition routes
 namespace {
 
+template<typename C>
+class lw_shared_container {
+public:
+    using iterator = C::iterator;
+    using value_type = C::value_type;
+
+    explicit lw_shared_container(C&& c)
+      : c_{ss::make_lw_shared<C>(std::move(c))} {}
+
+    iterator begin() const { return c_->begin(); }
+    iterator end() const { return c_->end(); }
+
+private:
+    ss::lw_shared_ptr<C> c_;
+};
```

Comment:
nit: should this be in some utility folder somewhere?  Could this be useful for other refactoring situations?

Reply:
I was wondering that; I haven't seen anything similar anywhere. I expect it will be useful again in this file, but perhaps also pandaproxy/rest. It can be moved later, I guess.